### DISCUSSION TITLE
Individual Vault Item Encryption Feature

### DISCUSSION THREAD

**Discussion by netboy3** [dani-garcia/vaultwarden#3976] [General]
Bitwarden merged two weeks ago PR's for both clients and the server that implement "Individual Vault Item Encryption Feature" See:
https://github.com/bitwarden/server/pull/3256
https://github.com/bitwarden/clients/pull/6241
And shows up in 2023.9.2. Is vaultwarden going to implement such changes too, and if not, will there be a possible issue with 2023.9.2 based clients?

**Comment by stefan0xC** [comment]
It will not show up in `2023.9.2` because the feature flag was disabled in https://github.com/bitwarden/clients/commit/0d371061a8f9dd51d44c0cfa8c105f47fd424e94

The client implements [an additional safeguard that also checks for the minimum server version](https://github.com/bitwarden/clients/blob/0d371061a8f9dd51d44c0cfa8c105f47fd424e94/libs/common/src/vault/services/cipher.service.ts#L1293-L1300), so there should not be an issue with `web-v2023.9.2`. However, we will probably not update the web-vault to `2023.9.2` because Vaultwarden has not implemented the server-side changes yet (and besides the merging of this feature branch there are [no other changes in `web-v2023.9.2`](https://github.com/bitwarden/clients/compare/web-v2023.9.1...web-v2023.9.2)). 

We will probably implement this feature eventually so Vaultwarden stays compatible with future versions of the web-vault that have this feature flag enabled (because other features will probably depend on individual item encryption some day).
